### ISSUE TITLE
Allow procs to be put under headings in docgen

### ISSUE THREAD

**Issue by dom96** [nim-lang/Nim#10774]
As discussed in #10752, I believe it would be really advantageous to enable us to put procedures under headings.

Something like:

```nim
## Main procedures
## -----------------
## 
## These are the main procedures in the module, the rest are irrelevant. Here is how you use them...

proc toMD5() ...

## Less useful procedures
## -----------------------
##
## Don't use these unless you have no other choice
```

**Comment by awr1** [comment]
Cool idea, some things to add:

- Why just procs? What about types too? Headings should probably be exclusive to one type of content, e.g. XYZ heading can only contain either types or procs, but not both
- Docgen appears to like to reorder the way types, procs, etc. are listed when `{.experimental: "codeReordering".}` is enabled, but for this I think Docgen should preserve the original "user order" if that pragma is active.

**Comment by krux02** [comment]
This conflicts with the constraint that procedures can only be used when they are at least declared first. Procedures need to be defined in the order how they are used. You can't just change the order for headings. 

I think it should be possible to split the documentation and the implementaiton into two different files. Something like ``mymodule.nim`` and ``mymodule.rst``. The documentation generator can then merges the information from the two files, for example use headings from the rst file. Whenever I write more documentation than the basic usage, I prefer to write it in an actual markup language, not as port of the comments in the sourcecode.

**Comment by awr1** [comment]
I really like the idea of seperate doc files. Putting documentation in with the source "bloats" the source file (unless you use things like code folding in your editor, which I'm not a huge fan of).

I think this also might pave the way for more flexibility with how we can document our programs. The way docgen works right now tends to be a bit unnecessarily verbose with the way type sections are rendered, especially in the way it shows private fields (which generally shouldn't be available in the docs)

**Comment by dom96** [comment]
Yeah, I'm happy with that too. We can have a mix too, like only specify the high-level docs in the separate file but keep the proc-specific docs inside the .nim file.
